tests/endpoints/bookstore_grpc/proto/v1/bookstore.proto

```diff
@@ -172,6 +165,17 @@ message GetShelfRequest {
   int64 shelf = 1;
 }
 
+// A shelf resource.
+message Shelf {
+  // A unique shelf id.
+  int64 id = 1;
+  // A theme of the shelf (fiction, poetry, etc).
+  string theme = 2;
+
+  google.protobuf.Any any = 3;
```

Comment:
What does the bookstore gRPC server do with this Any? I don't see any code in this PR that changes the bookstore gRPC server.

I would prefer if you could add some validation on the backend to ensure it receives the correct type. For example, in your integration test you send a `type.googleapis.com/endpoints.examples.bookstore.Book`. So have the bookstore server verify it can unmarshal the `Any` into a `Book`.

Reply:
Done.

Reply:
It is different problems:  from JSON to proto.Any on the request path, and from proto.Any to JSON on the response path. 

We need to test both.

Reply:
The test added covers json -> any on the request path and any -> json on the response path.